pyfr/plugins/fwh.py

```diff
@@ -45,29 +95,28 @@ def __init__(self, intg, cfgsect, suffix=None, *args, **kwargs):
         self._pidx = privars.index('p')
         self.consts = self.cfg.items_as('constants', float)
 
-        self.qinf = {k: npeval(self.cfg.getexpr(cfgsect, k), self.consts)
-                     for k in privars}
-        self.uinf = np.array([[self.qinf[k]] for k in 'uvw'[:self.ndims]])
+        self.qinf = qinf = {k: npeval(self.cfg.getexpr(cfgsect, k), 
+                                      self.consts) for k in privars}
+        self.uinf = np.array([[qinf[k]] for k in 'uvw'[:self.ndims]])
 
         if self.incomp:
-            self.qinf['rho'] = self.cfg.getfloat(cfgsect, 'rho')
-            self.qinf['c'] = self.cfg.getfloat(cfgsect, 'c')
+            qinf['rho'] = self.cfg.getfloat(cfgsect, 'rho')
+            qinf['c'] = self.cfg.getfloat(cfgsect, 'c')
         else:
             gamma = self.consts['gamma']
-            self.qinf['c'] = (gamma * self.qinf['p'] / self.qinf['rho'])**0.5
+            qinf['c'] = (gamma * qinf['p'] / qinf['rho'])**0.5
             self._ridx = privars.index('rho')
 
-        self.qinf['M'] = np.array([self.qinf[k] / self.qinf['c']
-                                   for k in 'uvw'[:self.ndims]])
-        Minf = np.linalg.norm(self.qinf['M'])
-        if Minf >= 1:
-            raise ValueError('FWH farfield Mach number greater than 1')
+        qinf['M'] = np.array([qinf[k] / qinf['c'] for k in 'uvw'[:self.ndims]])
 
         # Initialise surface data
         ele_map = intg.system.ele_map
         self.emap = {k: i for i, k in enumerate(ele_map)}
-        self.ele_surface, _ = self._surf_region(intg)
-        self._init_surf(ele_map, Minf)
+        ele_surf, _ = self._surf_region(intg)
+
+        surf_list = [(etype, eidx, fidx) for _, etype, fidx, eidx in ele_surf]
```

Comment:
Odd ordering (element type, list of indices, face number).  eidx should be plural and the ordering from ele_surf is more natural.

Reply:
I was following the ordering I was getting from `system.mesh.bcon.get(suffix, [])` but I've now switched to (element type, list of indices, face number). I'm reordering in `FluidForceIntegrator` as system is already being passed to the constructor, is this alright or would reordering in the `FluidForcePlugin` constructor before initialising the integrator be better?